readthedocs/rtd_tests/tests/test_doc_building.py

```diff
@@ -89,6 +140,54 @@ def test_normal_execution(self):
             'exit_code': 0,
         })
 
+    def test_record_command_as_success(self):
+        self.mocks.configure_mock('process', {
+            'communicate.return_value': (b'This is okay', '')
+        })
+        type(self.mocks.process).returncode = PropertyMock(return_value=1)
+
+        build_env = LocalBuildEnvironment(
+            version=self.version,
+            project=self.project,
+            build={'id': DUMMY_BUILD_ID},
+        )
+
+        with build_env:
+            build_env.run('echo', 'test', record_as_success=True)
+        self.assertTrue(self.mocks.process.communicate.called)
+        self.assertTrue(build_env.done)
+        self.assertFalse(build_env.successful)
+        self.assertEqual(len(build_env.commands), 1)
+        self.assertEqual(build_env.commands[0].output, u'This is okay')
+
+        # api() is not called anymore, we use api_v2 instead
+        self.assertFalse(self.mocks.api()(DUMMY_BUILD_ID).put.called)
+        # The command was saved
+        command = build_env.commands[0]
+        self.mocks.mocks['api_v2.command'].post.assert_called_once_with({
+            'build': DUMMY_BUILD_ID,
+            'command': command.get_command(),
+            'description': command.description,
+            'output': command.output,
+            'exit_code': 0,
+            'start_time': command.start_time,
+            'end_time': command.end_time,
+        })
+        self.mocks.mocks['api_v2.build']().put.assert_called_with({
+            'id': DUMMY_BUILD_ID,
+            'version': self.version.pk,
+            'success': False,
+            'project': self.project.pk,
+            'setup_error': u'',
+            'length': mock.ANY,
+            'error': '',
+            'setup': u'',
+            'output': u'',
+            'state': u'finished',
+            'builder': mock.ANY,
+            'exit_code': 1,
```

Comment:
After the change I did the other day, this should probably be 0 now.

Reply:
A lot of tests change with that p: